### PR TITLE
Elasticsearch Paginator failed to calculate total for  versions >= 7 because the total is an object in the new versions.

### PR DESCRIPTION
**Pull request by nicolae-stelian**
ApiPlatform\Core\Bridge\Elasticsearch\DataProviderPaginator::getTotalItems is failing to calculate correct total items for elasticsearch version >= 7

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

### CODE REVIEW

**Comment by nicolae-stelian** [comment]
Hi @dunglas @meyerbaptiste !
Please, when have time review the change. I write test, but I change some code in the test class. 
Put the constants limit and offset as a class variables and not as integers, and refactoring something in getPaginator because for elasticsearch7 I need to create another type of document (with total as object).

File: src/Bridge/Elasticsearch/DataProvider/Paginator.php [hunk 1]
```diff
@@ -68,7 +68,15 @@ public function getLastPage(): float
      */
     public function getTotalItems(): float
     {
-        return isset($this->documents['hits']['total']) ? (float) $this->documents['hits']['total'] : 0.;
+        // for elastic search version > 7.0.0
+        if (isset($this->documents['hits']['total']['value'])) {
+            return (float) $this->documents['hits']['total']['value'];
+        }
+        // for elastic search old versions
+        if (isset($this->documents['hits']['total'])) {
```
Comment (dunglas) [inline src/Bridge/Elasticsearch/DataProvider/Paginator.php]: Could you keep the ternary?

Of you can also write `return (float) ($this->documents['hits']['total'] ?? 0.)`;

File: tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php [hunk 2]
```diff
@@ -76,6 +76,9 @@ class PaginatorTest extends TestCase
         ],
     ];
 
+    protected $offset = 4;
```
Comment (dunglas) [inline tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php]: Both parameters should be class constants.

File: tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php [hunk 3]
```diff
@@ -149,20 +168,31 @@ function (array $document): Foo {
...
+        foreach ($documents['hits']['hits'] as $document) {
             $denormalizerProphecy
                 ->denormalize($document, Foo::class, ItemNormalizer::FORMAT, [AbstractNormalizer::ALLOW_EXTRA_ATTRIBUTES => true])
                 ->willReturn($this->denormalizeFoo($document['_source']));
         }
+        return $denormalizerProphecy;
+    }
 
-        return new Paginator($denormalizerProphecy->reveal(), self::DOCUMENTS, Foo::class, $limit, $offset);
+    protected function createPaginator(int $limit, int $offset, $documents, $denormalizerProphecy)
```
Comment (dunglas) [inline tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php]: Just copy/paste this line to keep the tests explicit.

File: src/Bridge/Elasticsearch/DataProvider/Paginator.php (hunk 1 above)
Comment (nicolae-stelian) [reply src/Bridge/Elasticsearch/DataProvider/Paginator.php]: Ok. I write using ternary operator.

File: tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php (hunk 2 above)
Comment (nicolae-stelian) [reply tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php]: Ok. I put this params as class constants.

File: tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php (hunk 3 above)
Comment (nicolae-stelian) [reply tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php]: I remove createPaginator function , and ai use new Paginator explicit in the tests.

**Pull request closed without merging**